upstairs/src/lib.rs

```diff
@@ -1950,15 +2063,15 @@ impl GuestWork {
                 if let Some(data) = data {
```

Comment:
I'm not 100% sure we can't have both an error and something in the data field.  Is that prevented somewhere?  If this job is an error, then I don't think we want to copy whatever data it returned.

Reply:
It's prevented and tested for as of 2700eb6facd2e3edd39d6ca937504ed0606af5fe